server/service/src/ledger_fix/fixes/mod.rs

```diff
@@ -117,37 +128,73 @@ fn create_inventory_adjustment(
     adjustment: f64,
     stock_line_id: &str,
 ) -> Result<(), LedgerFixError> {
-    let Some(StockLineLedgerRow {
-        item_id,
-        store_id,
-        datetime,
-        ..
-    }) = stock_line_ledger_row.map(|r| r.clone())
+    let Some(stock_line) =
+        StockLineRowRepository::new(connection).find_one_by_id(&stock_line_id)?
     else {
-        return LedgerFixError::other("Ledger line should exist for adjustment");
+        return LedgerFixError::other("Stock line not found");
+    };
+
+    let store_id = stock_line.store_id;
+
+    let datetime = match stock_line_ledger_row {
+        Some(r) => r.datetime.clone(),
+        None => {
```

Comment:
I really just wanted to use diesel directly here 😢. I just want to query

```sql
select * from invoice
join invoice_line on invoice_line.invoice_id = invoice.id
where invoice_line.stock_line_id = :param
order by invoice.finalised_datetime asc
limit 1;
```

But changing the repos would add to the return types, change the boxed queries, blah blah blah...